### PR TITLE
Add Alpine Linux CI

### PR DESCRIPTION
**Pull request by z3ntu**

### CODE REVIEW

**Comment by Saviq** [comment]
[copypasta from #136]

@z3ntu please make sure this works for you locally first, I tried this and got:

```
2019-09-08 11:10:07 Allocating lxd:alpine-3.10...
2019-09-08 11:10:08 Waiting for lxd container spread-161-alpine-3-10 to have an address...
2019-09-08 11:10:10 SSH install command is: exec spread-161-alpine-3-10 -- dnf --assumeyes install openssh-server
2019-09-08 11:10:10 Command output:
2019-09-08 11:10:11 Cannot allocate lxd:alpine-3.10: exit status 127
```

Spread incorrectly tries to use `dnf` to install ssh on the alpine instance. Partly because of snapcore/spread#42, partly because of snapcore/spread#48, which is included in our spread (see [our `.travis.yml`](https://github.com/MirServer/wlcs/blob/master/.travis.yml#L29).

**Comment by z3ntu** [comment]
I don't know how I can run this locally, I've been hoping that I can run it on Travis and fix it like that. 

So what can I do about spread trying to use dnf? I haven't used spread at all and I can't spot anything useful in the readme of that project

**Comment by Saviq** [comment]
> I don't know how I can run this locally, I've been hoping that I can run it on Travis and fix it like that.

`snap install spread` and `spread lxd:alpine-3.10:spread/build/alpine:gcc` should DTRT.

> So what can I do about spread trying to use dnf? I haven't used spread at all and I can't spot anything useful in the readme of that project

This would be a bug in spread itself, @RAOF is maintaining a version of it that has a couple of fixes for Mir, and a fix for Alpine would have to be added to that.

**Comment by RAOF** [comment]
Yeah, this is my bad. I've made the changes to our fork of spread that should add support for Alpine, I just need to remember how to actually make it build *my code*.

**Comment by RAOF** [comment]
Point your .travis.yml at https://cooperteam.net/spread.snap ; this now supports Alpine. This PR fails (you're missing the build depends) but should work after that's fixed!

**Comment by z3ntu** [comment]
I can't get spread to run in my Ubuntu 18.04 VM (and as Travis doesn't like my PRs for some reason it also doesn't run there):
image

**Comment by AlanGriffiths** [comment]
bors try

**Comment by AlanGriffiths** [comment]
> Travis doesn't like my PRs for some reason

Sorry, this isn't personal, but the build environment has access we cannot allow to arbitrary PRs. (I.e. it's team members only)

**Comment by z3ntu** [comment]
Weird, the previous PR #136 ran on Travis twice... https://travis-ci.org/MirServer/wlcs/builds/582256729 and https://travis-ci.org/MirServer/wlcs/builds/582257338

**Comment by AlanGriffiths** [comment]
OK, I guess I misundersood what's happening. I'd guess your changes are confusing Travis.

**Comment by z3ntu** [comment]
You were right, https://travis-ci.org/MirServer/wlcs/requests shows `Build config did not create any jobs.` for some reason. I've got Travis running now

**Comment by z3ntu** [comment]
CI should succeed now and is ready from my side

File: .travis.yml [hunk 1]
```diff
@@ -29,9 +29,9 @@ stages:
 install:
   - sudo adduser $USER lxd
   - sudo apt remove --assume-yes lxd
-  - wget https://cooperteam.net/spread_2017.09.21_amd64.snap
+  - wget https://cooperteam.net/spread.snap
```
Comment (AlanGriffiths) [inline .travis.yml]: @RAOF is that right?

Comment (RAOF) [reply .travis.yml]: Yup.

**Comment by RAOF** [comment]
Yup, that looks good.
bors r+

**Comment by RAOF** [comment]
Travis exploded while setting up lxd
bors r+

File: spread/build/ubuntu/task.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 summary: Build (on Ubuntu)
-systems: [-fedora-*]
+systems: [-fedora-*, -alpine-*]
```
Comment (z3ntu) [inline spread/build/ubuntu/task.yaml]: Should this maybe be changed to something like `systems: [ubuntu-*]` ?

Comment (RAOF) [reply spread/build/ubuntu/task.yaml]: Yes, probably :)

**Comment by z3ntu** [comment]
Thanks!

**Pull request merged**